src/visualizers/FormulaGrid.ts

```diff
@@ -0,0 +1,794 @@
+    **/
+    fillOrder: {
+        default: FillOrder.By_Rows,
+        type: ParamType.ENUM,
+        from: FillOrder,
+        displayName: 'Fill order',
+        required: true,
+    },
+    /** md
+- Path formula: A formula giving the _x_ and _y_ coordinates of the _k_-th
+  cell to be visited as the visualization is drawn. It is only shown/used
+  if Fill order is 'Custom'. The value may be a two-element array giving the
+  _x_ and _y_ coordinates in that order, or a plain object with keys `x`
+  and `y` giving the coordinates. The formula may use any of the variables
+  listed in the description above, where the inputs _x_, _y_, and _s_ are
+  calculated as if the fill order were the default 'By_Rows'. This latter
+  convention is useful if you want to tweak the default fill order.
+    **/
+    pathFormula: {
+        default: new MathFormula(path[FillOrder.By_Rows], formulaSymbols),
+        type: ParamType.FORMULA,
+        symbols: formulaSymbols,
+        displayName: 'Path formula',
+        required: false,
+        description:
+            'A formula giving the coordinates of the k-th cell '
+            + 'to be filled. '
+            + formulaDescription
+            + ' In this case, inputs x, y, and s are computed as if the path '
+            + 'were By_Rows.',
+        hideDescription: true,
+        visibleDependency: 'fillOrder',
+        visibleValue: FillOrder.Custom,
+    },
+    /** md
+- Background color: The color of the background.
+    **/
+    backgroundColor: {
+        default: '#FFFFFF',
+        type: ParamType.COLOR,
+        displayName: 'Background color',
+        required: true,
+    },
+    /** md
+- Speed: Cells filled per animation frame.
+    **/
+    speed: {
+        default: 32,
+        type: ParamType.INTEGER,
+        displayName: 'Speed',
+        required: false,
+        description: 'Cells filled per animation frame',
+        validate: function (s: number, status: ValidationStatus) {
+            status.mandate(s > 0, 'must be positive')
+        },
+    },
+    /** md
+- Fill formula: This parameter specifies the core behavior of Formula Grid,
+  directing what is drawn in each cell as the visualization reaches it.
+  As such, it is documented fully in the overview above.
+    **/
+    fillFormula: {
+        default: new MathFormula('a', formulaSymbols),
+        type: ParamType.FORMULA,
+        symbols: formulaSymbols,
+        displayName: 'Fill formula',
+        required: true,
+        description:
+            'A formula giving the color or array of colors to fill the '
+            + 'cell with, or a plain object associating with some shape '
+            + 'names the color or array of colors to draw them with. Allowed '
+            + 'keys are rectangle, square, ellipse, circle, hexagon, '
+            + 'triangle, text, or mouseover; the values for the last two '
+            + '"shapes" will be (converted to) strings, not colors. '
+            + formulaDescription,
```

Comment:
I'm not sure how to improve this pop up description, but the use of this cell is very complicated and a tooltip can't possibly do it justice.  Maybe it should just say something like "A formula giving the color of the cell, or a more complicated formula describing shapes, text and mouseover information for the cell."  Because they need to visit the docs, maybe the only goal here is to describe the function, not the format.  Perhaps this comment could even apply to some of the other tooltips also.

Reply:
Deferring this for in-person conversation.